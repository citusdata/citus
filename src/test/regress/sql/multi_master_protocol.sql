--
-- MULTI_MASTER_PROTOCOL
--
-- Tests that check the metadata returned by the master node.


SET citus.next_shard_id TO 740000;


SELECT part_storage_type, part_key, part_replica_count, part_max_size,
	   part_placement_policy FROM master_get_table_metadata('lineitem');

SELECT * FROM master_get_table_ddl_events('lineitem') order by 1;

SELECT * FROM master_get_new_shardid();

SELECT * FROM master_get_active_worker_nodes();
