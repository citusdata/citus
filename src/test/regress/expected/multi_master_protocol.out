--
-- MULTI_MASTER_PROTOCOL
--
-- Tests that check the metadata returned by the master node.
ALTER SEQUENCE pg_catalog.pg_dist_shardid_seq RESTART 740000;
ALTER SEQUENCE pg_catalog.pg_dist_jobid_seq RESTART 740000;
SELECT part_storage_type, part_key, part_replica_count, part_max_size,
	   part_placement_policy FROM master_get_table_metadata('lineitem');
 part_storage_type |  part_key  | part_replica_count | part_max_size | part_placement_policy 
-------------------+------------+--------------------+---------------+-----------------------
 t                 | l_orderkey |                  2 |        307200 |                     2
(1 row)

SELECT * FROM master_get_table_ddl_events('lineitem');
                                                                                                                                                                                                                                                                         master_get_table_ddl_events                                                                                                                                                                                                                                                                         
-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 CREATE TABLE public.lineitem (l_orderkey bigint NOT NULL, l_partkey integer NOT NULL, l_suppkey integer NOT NULL, l_linenumber integer NOT NULL, l_quantity numeric(15,2) NOT NULL, l_extendedprice numeric(15,2) NOT NULL, l_discount numeric(15,2) NOT NULL, l_tax numeric(15,2) NOT NULL, l_returnflag character(1) NOT NULL, l_linestatus character(1) NOT NULL, l_shipdate date NOT NULL, l_commitdate date NOT NULL, l_receiptdate date NOT NULL, l_shipinstruct character(25) NOT NULL, l_shipmode character(10) NOT NULL, l_comment character varying(44) NOT NULL)
 CREATE INDEX lineitem_time_index ON public.lineitem USING btree (l_shipdate)
 ALTER TABLE public.lineitem ADD CONSTRAINT lineitem_pkey PRIMARY KEY (l_orderkey, l_linenumber)
(3 rows)

SELECT * FROM master_get_new_shardid();
 master_get_new_shardid 
------------------------
                 740000
(1 row)

SELECT * FROM master_get_local_first_candidate_nodes();
 node_name | node_port 
-----------+-----------
 localhost |     57638
 localhost |     57637
(2 rows)

SELECT * FROM master_get_round_robin_candidate_nodes(1);
 node_name | node_port 
-----------+-----------
 localhost |     57638
 localhost |     57637
(2 rows)

SELECT * FROM master_get_round_robin_candidate_nodes(2);
 node_name | node_port 
-----------+-----------
 localhost |     57637
 localhost |     57638
(2 rows)

SELECT * FROM master_get_active_worker_nodes();
 node_name | node_port 
-----------+-----------
 localhost |     57638
 localhost |     57637
(2 rows)

